### PR TITLE
partially enable type checking for .models

### PR DESCRIPTION
**Pull request by pmeier**

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -4,15 +4,19 @@ files = torchvision
...
 pretty = True
 
-;[mypy-torchvision.datasets.*]
+[mypy-torchvision.io.*]
 
-;ignore_errors = True
+ignore_errors = True
 
-[mypy-torchvision.io.*]
+[mypy-torchvision.models.detection.*]
```
Comment (pmeier) [inline mypy.ini:11]: In total `mypy` found 86 existing errors in `detection`, `densenet`, and `quantization`. I'm ignoring them here to get a clean run and will fix them in separate follow-up PRs.

File: torchvision/models/googlenet.py [hunk 2]
```diff
@@ -193,8 +194,7 @@ def _forward(self, x):
         return x, aux2, aux1
 
     @torch.jit.unused
-    def eager_outputs(self, x, aux2, aux1):
-        # type: (Tensor, Optional[Tensor], Optional[Tensor]) -> GoogLeNetOutputs
+    def eager_outputs(self, x: Tensor, aux2: Tensor, aux1: Optional[Tensor]) -> Union[GoogLeNetOutputs, Tensor]:
```
Comment (pmeier) [inline torchvision/models/googlenet.py:197]: Not sure if I can use `Union` when the method is decorated with `@torch.jit.unused`. If not, we can only ignore this error.

Comment (fmassa) [reply torchvision/models/googlenet.py:197]: I'm not 100% sure. Tests seem to pass, so it should be fine?

cc @eellison to double-check

**Review by fmassa — APPROVED** [review]
Looks good to me, conditioned on @eellison answer to the `Union` part.

**Comment by pmeier** [comment]
@fmassa I'll rebase after @eellison s review.

File: torchvision/models/googlenet.py (hunk 2 above)
Comment (eellison) [reply torchvision/models/googlenet.py:197]: Yea, the type does get used so I don't think you can use union. Can try `Any` maybe ?

Comment (fmassa) [reply torchvision/models/googlenet.py:197]: Thanks for confirming @eellison , although I'm a bit worried why torchscript tests passed then.

I would say just keep the old type annotation as before, and forget the `Tensor` return type?

Comment (eellison) [reply torchvision/models/googlenet.py:197]: Actually i could be wrong... let me double check sorry

Comment (eellison) [reply torchvision/models/googlenet.py:197]: Okay, it does look at annotations.

Comment (pmeier) [reply torchvision/models/googlenet.py]: I went for the ignore for now.

**Pull request merged**